logprep/processor/domain_resolver/rule.py

```diff
@@ -34,8 +34,10 @@ def __init__(self, filter_rule: FilterExpression, domain_resolver_cfg: dict):
             self._output_field = "resolved_ip"
 
     def __eq__(self, other: "DomainResolverRule") -> bool:
-        return (other.filter == self._filter) and (
-            self._source_url_or_domain == other.source_url_or_domain
+        return (
+            (other.filter == self._filter)
+            and (self._source_url_or_domain == other.source_url_or_domain)
+            and (self._output_field == other.output_field)
```

Comment:
please use the builtin `all` function for performance reasons. it is implemented in C